### PR TITLE
MNT Add changes for compatibility with CMS5 and testing with framework

### PR DESCRIPTION
**Pull request by sabina-talipova**
### Description
- Replace old method `first` to new method `record`;
- Remove `__config.yml` from root folder and add empty `config.yml` in `_config` folder;
- Add jobs to GA to test together with `framework`;
- Add test cases for this module;
- Remove `seek` method as deprecated and removed from framework ORM `Query::class`
`Query now implements IteratorAggregate instead of Iterator. This means seek() and other iterator methods are no longer available on this class and its subclasses. Use getIterator() instead.` [See changelog](https://docs.silverstripe.org/en/5/changelogs/5.0.0/#api-orm)

### Parent issue
- https://github.com/silverstripe/silverstripe-sqlite3/issues/68

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,3 +14,37 @@ jobs:
...
     if: (github.event_name == 'schedule' && github.repository_owner == 'silverstripe') || (github.event_name != 'schedule')
     uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    with:
+      # set phpunit to false to prevent automatic generation of mysql phpunit jobs
+      phpunit: false
+      extra_jobs: |
+        - php: 8.1
+          db: sqlite3
+          phpunit: true
+          composer_args: --prefer-lowest --prefer-source
```
Comment (emteknetnz) [inline .github/workflows/ci.yml:24]: We don't need `--prefer-source` for any of these - we'll add a new input `preserve_vendor_tests` in https://github.com/silverstripe/gha-ci/pull/58/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR42

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -14,3 +14,37 @@ jobs:
     # Only run cron on the silverstripe account
     if: (github.event_name == 'schedule' && github.repository_owner == 'silverstripe') || (github.event_name != 'schedule')
     uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    with:
+      # set phpunit to false to prevent automatic generation of mysql phpunit jobs
+      phpunit: false
+      extra_jobs: |
+        - php: 8.1
+          db: sqlite3
+          phpunit: true
```
Comment (emteknetnz) [inline .github/workflows/ci.yml:23]: We don't need to specify `phpunit: true` - it's true by default https://github.com/silverstripe/gha-ci/blob/1/.github/workflows/ci.yml#L38

File: phpcs.xml.dist [hunk 3]
```diff
@@ -10,3 +10,13 @@
...
+<phpunit bootstrap="vendor/silverstripe/framework/tests/bootstrap.php" colors="true">
+    <testsuites>
+        <testsuite name="sqlite3">
+            <directory>tests</directory>
+        </testsuite>
+        <testsuite name="framework">
+            <directory>vendor/silverstripe/framework/tests/php</directory>
+        </testsuite>
+    </testsuites>
+</phpunit>
```
Comment (emteknetnz) [inline phpcs.xml.dist:22]: Add a newline at the the end of the file

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sabina-talipova) [reply .github/workflows/ci.yml:24]: DONE

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sabina-talipova) [reply .github/workflows/ci.yml:23]: DONE

File: phpcs.xml.dist (hunk 3 above)
Comment (sabina-talipova) [reply phpcs.xml.dist:22]: DONE

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -14,3 +14,20 @@ jobs:
...
     uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    with:
+      # set phpunit to false to prevent automatic generation of mysql phpunit jobs
+      phpunit: false
+      extra_jobs: |
+        - php: 8.1
+          db: sqlite3
+          composer_args: --prefer-lowest
+          phpunit_suite: framework
+          preserve_vendor_tests: true
```
Comment (maxime-rainville) [inline .github/workflows/ci.yml:25]: I've added the `preserve_vendor_tests` statement here to match an upstream change on the CI workflow.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -13,4 +13,21 @@ jobs:
     name: CI
     # Only run cron on the silverstripe account
     if: (github.event_name == 'schedule' && github.repository_owner == 'silverstripe') || (github.event_name != 'schedule')
-    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    uses: maxime-rainville/gha-ci/.github/workflows/ci.yml@v1
```
Comment (maxime-rainville) [inline .github/workflows/ci.yml:16]:
```suggestion
    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
```

Need to switch this back before merging.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -13,4 +13,21 @@ jobs:
     name: CI
     # Only run cron on the silverstripe account
     if: (github.event_name == 'schedule' && github.repository_owner == 'silverstripe') || (github.event_name != 'schedule')
-    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
+    uses: maxime-rainville/gha-ci/.github/workflows/ci.yml@1
```
Comment (maxime-rainville) [inline .github/workflows/ci.yml:16]:
```suggestion
    uses: silverstripe/gha-ci/.github/workflows/ci.yml@v1
```

Revert to this before merge.

File: code/SQLite3Query.php [hunk 7]
```diff
@@ -72,13 +63,10 @@ public function numRecords()
...
 
-    #[\ReturnTypeWillChange]
-    public function getIterator()
+    public function getIterator(): Traversable
     {
-        if ($data = $this->handle->fetchArray(SQLITE3_ASSOC)) {
-            return $data;
-        } else {
-            return false;
+        while ($data = $this->handle->fetchArray(SQLITE3_ASSOC)) {
```
Comment (maxime-rainville) [inline code/SQLite3Query.php:68]: This was initially only returning the first row. I swap it to a `while` statement so it would iterate over the data.

**Review by maxime-rainville — COMMENTED** [review]
@sabina-talipova I took some liberties with your PR to debug the CI set up ... but the build is running now.

It's not quite green thought. I didn't dig into what is failing, but there is a light at the end of this tunnel ... I think.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (maxime-rainville) [reply .github/workflows/ci.yml:25]: @emteknetnz I think you do have to specify it if you define in as false on the level up.

**Comment by maxime-rainville** [comment]
This Framework PR should fix the broken build https://github.com/silverstripe/silverstripe-framework/pull/10713

File: composer.json [hunk 8]
```diff
@@ -30,5 +31,11 @@
...
     },
     "minimum-stability": "dev",
-    "prefer-stable": true
+    "prefer-stable": true,
+	"config": {
+		"allow-plugins": {
+			"composer/installers": true,
+			"silverstripe/vendor-plugin": true
+		}
+	}
```
Comment (emteknetnz) [inline composer.json:40]:
```suggestion
```
Don't need these here, we have allow-plugins in installer + sink, but not other modules

File: .github/workflows/ci.yml (hunk 6 above)
Comment (emteknetnz) [reply .github/workflows/ci.yml:16]: This still needs to be reverted.  The corresponding gha-ci pr has been merged and released

**Review by emteknetnz — CHANGES_REQUESTED** [review]

**Comment by maxime-rainville** [comment]
The failure on the prefer lowest build is being cause by a deprecation warning that got fixed in masterminds/html5 v2.7.5

https://github.com/silverstripe/silverstripe-framework/pull/10714

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**